projects/common/src/time/time-duration.ts

```diff
@@ -1,13 +1,16 @@
 import { parse, toSeconds } from 'iso8601-duration';
+import { toString } from 'lodash-es';
 import { assertUnreachable } from '../utilities/lang/lang-utils';
 import { TimeUnit } from './time-unit.type';
 
 export class TimeDuration {
   private readonly millis: number;
+  private readonly militaryTime: Date;
 
   public constructor(public readonly value: number, public readonly unit: TimeUnit) {
     this.toUnitString(); // Fail if unrecognized TimeUnit
     this.millis = this.normalizeToMillis(isNaN(value) ? 0 : value, unit);
+    this.militaryTime = this.buildMilitaryTime();
```

Comment:
Avoid no-arg methods that rely on state. If these two calls had their order flipped, it would break.